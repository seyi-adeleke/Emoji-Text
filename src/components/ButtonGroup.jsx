import React from 'react';
import PropTypes from 'prop-types';

class ButtonGoup extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
    };
    this.handleEmojiChange = this.handleEmojiChange.bind(this);
  }


  handleEmojiChange(event) {
    event.preventDefault();
    this.props.getEmoji(event.target.value);
    this.setState({
      emoji: event.target.value,
    });
  }


  render() {
    const emojis = ['ğŸ˜‚', 'ğŸ˜˜', 'ğŸ˜¢', 'ğŸ™ˆ', 'âœŠ', 'âš¡', 'ğŸ’“', 'ğŸ’¦', 'ğŸ’…', 'ğŸ”ª', 'ğŸ”¥', 'ğŸ–•'];
    return (
      <div className="field is-grouped is-grouped-multiline has-text-centered">
        {
          emojis.map((emoji, index) => {
            return <p className="control" key={index}>
                      <button className="button"
                      value={emoji}
                      onClick={this.handleEmojiChange}>
                      {emoji}
                     </button>
                  </p>;
          })
        }
      </div>
    );
  }
}

ButtonGoup.propTypes = {
  getEmoji: PropTypes.string
};


export default ButtonGoup;
