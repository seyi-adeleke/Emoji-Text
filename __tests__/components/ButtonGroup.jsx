import React from 'react';
import { shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
// import Adapter from 'enzyme-adapter-react-15';

import ButtonGroup from '../../src/components/ButtonGroup.jsx';

// Enzyme.configure({ adapter: new Adapter() });

describe('Button Component', () => {
  it('should match the Button snapshot', () => {
    const component = mount(<ButtonGroup />);
    const tree = toJson(component);
    expect(tree).toMatchSnapshot();
  });
});
