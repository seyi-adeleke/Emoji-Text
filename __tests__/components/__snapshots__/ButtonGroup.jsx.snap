// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component should match the Button snapshot 1`] = `
<ButtonGoup>
  <div
    className="field is-grouped is-grouped-multiline has-text-centered"
  >
    <p
      className="control"
    >
      <button
        className="button"
        onClick={[Function]}
        value="ðŸ˜‚"
      >
        ðŸ˜‚
      </button>
    </p>
    <p
      className="control"
    >
      <button
        className="button"
        onClick={[Function]}
        value="ðŸ˜˜"
      >
        ðŸ˜˜
      </button>
    </p>
    <p
      className="control"
    >
      <button
        className="button"
        onClick={[Function]}
        value="ðŸ˜¢"
      >
        ðŸ˜¢
      </button>
    </p>
    <p
      className="control"
    >
      <button
        className="button"
        onClick={[Function]}
        value="ðŸ™ˆ"
      >
        ðŸ™ˆ
      </button>
    </p>
    <p
      className="control"
    >
      <button
        className="button"
        onClick={[Function]}
        value="âœŠ"
      >
        âœŠ
      </button>
    </p>
    <p
      className="control"
    >
      <button
        className="button"
        onClick={[Function]}
        value="âš¡"
      >
        âš¡
      </button>
    </p>
    <p
      className="control"
    >
      <button
        className="button"
        onClick={[Function]}
        value="ðŸ’“"
      >
        ðŸ’“
      </button>
    </p>
    <p
      className="control"
    >
      <button
        className="button"
        onClick={[Function]}
        value="ðŸ’¦"
      >
        ðŸ’¦
      </button>
    </p>
    <p
      className="control"
    >
      <button
        className="button"
        onClick={[Function]}
        value="ðŸ’…"
      >
        ðŸ’…
      </button>
    </p>
    <p
      className="control"
    >
      <button
        className="button"
        onClick={[Function]}
        value="ðŸ”ª"
      >
        ðŸ”ª
      </button>
    </p>
    <p
      className="control"
    >
      <button
        className="button"
        onClick={[Function]}
        value="ðŸ”¥"
      >
        ðŸ”¥
      </button>
    </p>
    <p
      className="control"
    >
      <button
        className="button"
        onClick={[Function]}
        value="ðŸ–•"
      >
        ðŸ–•
      </button>
    </p>
  </div>
</ButtonGoup>
`;
